deploy/terraform-vpn/main.tf

```diff
@@ -0,0 +1,107 @@
+  url = "http://ipv4.icanhazip.com"
+}
+
+locals {
+  client_ip           = chomp(data.http.client_ip.body)
+  client_cidr         = "${chomp(data.http.client_ip.body)}/32"
+  extra_cidrs         = concat([local.client_cidr], var.local_cidrs)
+  vpc_public_subnets  = [cidrsubnet(var.vpc_cidr, 4, 1)]
+}
+
+module "vpn_gateway" {
+  source = "terraform-aws-modules/vpn-gateway/aws"
+  version = "~> 2.0"
+
+  create_vpn_connection                     = true
+  vpn_connection_static_routes_only         = true
+  vpn_connection_static_routes_destinations = var.local_cidrs
+
+  vpn_gateway_id      = module.vpc.vgw_id
+  customer_gateway_id = aws_customer_gateway.main.id
+
+  vpc_id                       = module.vpc.vpc_id
+  vpc_subnet_route_table_ids   = module.vpc.public_route_table_ids
+  vpc_subnet_route_table_count = length(module.vpc.public_subnets)
+
+  tunnel1_inside_cidr   = var.tunnel1_inside_cidr
+  tunnel2_inside_cidr   = var.tunnel2_inside_cidr
+  tunnel1_preshared_key = var.tunnel1_psk
+  tunnel2_preshared_key = var.tunnel2_psk
+}
+
+resource "aws_customer_gateway" "main" {
+  bgp_asn    = 65000
+  ip_address = local.client_ip
+  type       = "ipsec.1"
+
+  tags = {
+    Name = var.name
+  }
+}
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "~> 2.0"
+
+  name = var.name
+  cidr = var.vpc_cidr
+  azs  = var.azs
+  public_subnets  = local.vpc_public_subnets
+  enable_nat_gateway = false
+  enable_vpn_gateway = true
+
+  tags = {
+    Name        = var.name
+  }
+}
+
+resource "local_file" "aws-vpn-client-env" {
+  sensitive_content = templatefile("kustomization/aws-vpn-client.env.tmpl", {
+    tunnel1_address=module.vpn_gateway.vpn_connection_tunnel1_address,
+    tunnel1_cgw_inside_address=module.vpn_gateway.vpn_connection_tunnel1_cgw_inside_address,
+    tunnel1_vgw_inside_address=module.vpn_gateway.vpn_connection_tunnel1_vgw_inside_address,
+    tunnel1_psk=var.tunnel1_psk,
+    tunnel2_address=module.vpn_gateway.vpn_connection_tunnel2_address,
+    tunnel2_cgw_inside_address=module.vpn_gateway.vpn_connection_tunnel2_cgw_inside_address,
+    tunnel2_vgw_inside_address=module.vpn_gateway.vpn_connection_tunnel2_vgw_inside_address,
+    tunnel2_psk=var.tunnel2_psk,
+    vpc_cidr=var.vpc_cidr,
+  })
+  filename          = "kustomization/aws-vpn-client.env"
+  file_permission   = "0600"
+}
+
+resource "local_file" "provider-yaml" {
+  content         = templatefile("kustomization/provider.yaml.tmpl", {
+    name=var.name,
+    region=var.region,
+    vpc_id=module.vpc.vpc_id,
+    subnet_id=module.vpc.public_subnets[0],
+    extra_cidrs=local.extra_cidrs,
+  })
+  filename        = "kustomization/provider.yaml"
+  file_permission = "0644"
+}
+
+resource "null_resource" "deploy" {
+  count = var.deploy_to_kubernetes ? 1 : 0
+
+  provisioner "local-exec" {
+    command = "[[ -n \"$AWS_ACCESS_KEY_ID\" ]] && [[ -n \"$AWS_SECRET_ACCESS_KEY\" ]] && kustomize build kustomization/ | envsubst | kubectl apply -f -"
```

Comment:
users need to install kustomize... it isn't present on most machines.  They'll only realize its missing after spending 15 minutes creating a VPC and VPN gateway.

Reply:
I changed this to use `kubectl apply -k` instead. This requires kubectl>=1.14, but I guess that's better than relying on kustomize.